discovery-client/src/test/groovy/io/micronaut/discovery/consul/ConsulMockConfigurationClientNativeSpec.groovy

```diff
@@ -58,19 +62,45 @@ class ConsulMockConfigurationClientNativeSpec extends Specification {
     ConsulConfigurationClient configClient = embeddedServer.applicationContext.getBean(ConsulConfigurationClient)
 
     void "test read and write key values with ConsulClient"() {
-        when:"A property is written"
+        when: "A property is written"
         def result = Flux.from(client.putValue("config/application/datasource.url", "mysql://blah")).blockFirst()
 
-        then:"The operation was successful"
+        then: "The operation was successful"
         result
 
-        when:"Properties are read"
+        when: "Properties are read"
         Flux.from(client.putValue("config/application/datasource.driver", "java.SomeDriver")).blockFirst()
+        Flux.from(client.putValue("config/application,h2/datasource.driver", "h2.OtherDriver")).blockFirst()
+
+        List<KeyValue> matchingKeyValues = Flux.from(client.readValues("config/application")).blockFirst()
+        List<KeyValue> exactKeyValues = Flux.from(client.readValues("config/application", false)).blockFirst()
+
+        then:
+        matchingKeyValues.size() == 3
+        matchingKeyValues.collect { it.key }.toSet() == Set.of("config/application/datasource.url", "config/application/datasource.driver", "config/application,h2/datasource.driver")
+        exactKeyValues.size() == 2
+        exactKeyValues.collect { it.key }.toSet() == Set.of("config/application/datasource.url", "config/application/datasource.driver")
+    }
+
+    void "test watch key values with ConsulClient"() {
```

Comment:
I'm honestly not sure about how to exactly test the Blocking Query behavior from Consul